File: 129. Sum Root to Leaf Numbers/Solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int res;
    public int sumNumbers(TreeNode root) {
        res = 0;
        dfs(root, 0);
        return res;
    }
    public void dfs(TreeNode curr, int total) {
        if (curr == null) return;
        if (curr.left == null && curr.right == null) {
            res += total + curr.val;
            return;
        }
        dfs(curr.left, (total + curr.val) * 10);
        dfs(curr.right, (total + curr.val) * 10);
    }
}

File: 129. Sum Root to Leaf Numbers/Solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        self.res = 0
        self.dfs(root, 0)
        return self.res
    def dfs(self, curr, total):
        if not curr: return
        if not curr.left and not curr.right:
            self.res += total + curr.val
            return
        self.dfs(curr.left, (total + curr.val) * 10)
        self.dfs(curr.right, (total + curr.val) * 10)